encord/objects/classification_instance.py

```diff
@@ -95,9 +100,64 @@ def _last_frame(self) -> Union[int, float]:
         else:
             return self._parent.number_of_frames
 
+    @property
+    def range_list(self) -> Ranges:
+        if self._use_range or (self._parent is not None and self._parent.data_type == DataType.AUDIO):
+            return self._range_manager.get_ranges()
+        else:
+            # TODO: Think about this
+            raise LabelRowError("No ranges available for this classification instance.")
+
     def is_assigned_to_label_row(self) -> bool:
         return self._parent is not None
 
+    def set_for_ranges(
+            self,
+            frames: Frames,
+            overwrite: bool,
+            created_at: Optional[datetime],
+            created_by: Optional[str],
+            confidence: float,
+            manual_annotation: bool,
+            last_edited_at: Optional[datetime],
+            last_edited_by: Optional[str],
+            reviews: Optional[List[dict]],
+    ):
+        new_range_manager = RangeManager(frame_class=frames)
+        conflicting_ranges = self._is_classification_already_present_on_range(new_range_manager.get_ranges())
+        if conflicting_ranges and not overwrite:
+            raise LabelRowError(
+                f"The classification '{self.classification_hash}' already exists "
+                f"on the ranges {conflicting_ranges}. "
+                f"Set 'overwrite' parameter to True to override."
+            )
+
+        new_range_manager.remove_ranges(conflicting_ranges)
+
+        ranges_to_add = new_range_manager.get_ranges()
+        for range_to_add in ranges_to_add:
+            self._check_within_range(range_to_add.end)
+
+        # TODO: Think about this and write comment
+        self._frames_to_data = {}
+        self._set_frame_and_frame_data(
+            frame=0,
+            overwrite=overwrite,
+            created_at=created_at,
+            created_by=created_by,
+            confidence=confidence,
+            manual_annotation=manual_annotation,
+            last_edited_at=last_edited_at,
+            last_edited_by=last_edited_by,
+            reviews=reviews,
+        )
```

Comment:
At this point, we know that this classification_instance should be using ranges, and not frames, so we remove all the frame_data, and only set the frame_data for the "first" frame, which will have frameData that applies to all ranges in this classification instance.